keras_cv/models/generative/stable_diffusion/stable_diffusion.py

```diff
@@ -283,6 +283,144 @@ def generate_image(
         decoded = ((decoded + 1) / 2) * 255
         return np.clip(decoded, 0, 255).astype("uint8")
 
+    def inpaint(
+        self,
+        prompt,
+        image,
+        mask,
+        num_resamples=1,
+        batch_size=1,
+        num_steps=25,
+        unconditional_guidance_scale=7.5,
+        diffusion_noise=None,
+        seed=None,
+    ):
+        """Generates an image based on encoded text.
+
+        The encoding passed to this method should be derived from
+        `StableDiffusion.encode_text`.
+
+        Args:
+            prompt: A string representing the prompt for generation.
+            image: Tensor of shape (`batch_size`, `image_height`, `image_width`,
+            3) with RGB values in [0, 255]. When the batch is omitted, the same
+            image will be used as the starting image.
+            mask: Tensor of shape (`batch_size`, `image_height`, `image_width`)
+            with binary values 0 or 1. When the batch is omitted, the same mask
+            will be used on all images.
+            batch_size: number of images to generate. Default: 1.
+            num_steps: number of diffusion steps (controls image quality).
+                Default: 25.
+            unconditional_guidance_scale: float controling how closely the image
+                should adhere to the prompt. Larger values result in more
+                closely adhering to the prompt, but will make the image noisier.
+                Default: 7.5.
+            diffusion_noise: Tensor of shape (`batch_size`, img_height // 8,
```

Comment:
start this description with (Optional) because the default is randomly generated, but we should still tell users tehy don't have to pass it.